GPflowOpt/bo.py

```diff
@@ -165,3 +167,20 @@ def inverse_acquisition(x):
             self._update_model_data(result.x, fx(result.x))
 
         return self._create_bo_result(True, "OK")
+
+    @contextmanager
+    def failsafe(self):
+        """
+        Context to provide a safe way for optimization. If a RuntimeError is generated, the data of the acquisition
+        object is saved to the disc in the current directory. This allows the data to be re-used (which makes sense
+        for expensive data).
+
+        The data can also be used to try to fit a GPflow model first (set sensible initial
+        hyperparameter values and hyperpriors) before retrying Bayesian Optimization again.
+        """
+        try:
+            yield
+        except RuntimeError as e:
+            print(id(e))
+            np.savez('failed_bopt_{0}'.format(id(e)), X=self.acquisition.data[0], Y=self.acquisition.data[1])
```

Comment:
shouldn't we catch exceptions generated here (disk full, permission problem, etc.)? We can't do anything with it of course, except inform the user of the original RunTimeError (otherwise that error will be hidden I assume by the new exception).